tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs

```diff
@@ -22,28 +24,43 @@ public void EncodeGeneratedPatterns<TPixel>(TestImageProvider<TPixel> provider)
         {
             using (Image<TPixel> image = provider.GetImage())
             {
-                provider.Utility.SaveTestOutputFile(image, "gif", new GifEncoder());
+                var encoder = new GifEncoder()
+                {
+                    // Use the palette quantizer without dithering to ensure results 
+                    // are consistant
+                    Quantizer = new PaletteQuantizer(false)
+                };
+
+                // Always save as we need to compare the encoded output.
+                provider.Utility.SaveTestOutputFile(image, "gif", encoder);
+            }
+
+            // Compare encoded result
+            string path = provider.Utility.GetTestOutputFileName("gif", null, true);
+            using (var encoded = Image.Load(path))
+            {
+                encoded.CompareToReferenceOutput(ValidatorComparer, provider, null, "gif");
```

Comment:
Does this fail if you reintroduce the mistake in `WriteApplicationExtension()`?

If yes, I'm happy to merge this!

Reply:
Yeah you end up with a black and white image